test/ruby/test_module.rb

```diff
@@ -563,6 +563,10 @@ def test_name
     assert_equal("Integer", Integer.name)
     assert_equal("TestModule::Mixin",  Mixin.name)
     assert_equal("TestModule::User",   User.name)
+
+    assert Integer.name.frozen?
```

Comment:
Could you use `assert_predicate Integer.name, :frozen?` etc. for the better failure messages?

Reply:
Done.